### DISCUSSION TITLE
Custom Style Setter

### DISCUSSION THREAD

**Discussion by thexamlguy** [AvaloniaUI/Avalonia#11004] [Q&A]
Hi,

I have been using this library https://github.com/codingseb/Avalonia.EventSetter to set events within a style setter. It all worked with Preview3 as indicated with a dependency to within EventSetter package. However, this seems to longer seems to work past Preview3. I am expecting this wasn't something originally supported with Avalonia and that it was recently fixed in the latest Avalonia version.

Is something I could do to get the EventSetter working again, or is this now by design?

**Comment by timunie** [comment]
@TheXamlGuy the library needs to be compiled for preview6 in order to use it with preview6. Another option is probably to use Avalonia.Behavios: https://github.com/AvaloniaUI/Avalonia.Xaml.Behaviors 

https://github.com/AvaloniaUI/Avalonia.Xaml.Behaviors/blob/master/samples/BehaviorsTestApplication/Views/Pages/EventTriggerBehaviorView.axaml for example

**Comment by thexamlguy** [comment]
Hi @timunie,

Cheers for your reply. I have compiled the library for preview6, specifically 11.0.999-cibuild0033226-beta even so. Unfortunately it doesn't work.

I will take a look at the Behaviours to see if it gives what I need.

Thanks,
Dan
